lms/djangoapps/certificates/api.py

```diff
@@ -484,49 +484,41 @@ def get_active_web_certificate(course, is_preview_mode=None):
     return None
 
 
-def get_certificate_template(course_key, mode, language):  # pylint: disable=unused-argument
+def get_certificate_template(course_key, mode, language):
     """
     Retrieves the custom certificate template based on course_key, mode, and language.
     """
-    org_id, template = None, None
     # fetch organization of the course
-    course_organization = get_course_organizations(course_key)
-    if course_organization:
-        org_id = course_organization[0]['id']
-
-    if org_id and mode:
-        template = CertificateTemplate.objects.filter(
-            organization_id=org_id,
-            course_key=course_key,
-            mode=mode,
-            is_active=True
-        )
-    # if don't template find by org and mode
-    if not template and org_id and mode:
-        template = CertificateTemplate.objects.filter(
-            organization_id=org_id,
-            course_key=CourseKeyField.Empty,
-            mode=mode,
-            is_active=True
-        )
-    # if don't template find by only org
-    if not template and org_id:
-        template = CertificateTemplate.objects.filter(
-            organization_id=org_id,
-            course_key=CourseKeyField.Empty,
-            mode=None,
-            is_active=True
-        )
-    # if we still don't template find by only course mode
-    if not template and mode:
-        template = CertificateTemplate.objects.filter(
-            organization_id=None,
-            course_key=CourseKeyField.Empty,
-            mode=mode,
-            is_active=True
-        )
+    org_id = get_course_organization_id(course_key)
+
+    # prepare template sets.
+    active_templates = CertificateTemplate.objects.filter(is_active=True)
+    org_templates = active_templates.filter(organization_id=org_id)
+    mode_templates = active_templates.filter(mode=mode)
+    org_and_mode_templates = org_templates & mode_templates
+    key_org_and_mode_templates = org_and_mode_templates.filter(course_key=course_key)
 
-    return template[0].template if template else None
+    # Create list for easy operation, with template sets ordered by priority to return 
+    # e.g. a key_org_and_mode_templates should be returned before an org_and_mode_template
+    template_sets = [key_org_and_mode_templates, org_and_mode_templates, org_templates, mode_templates]
+    
+    # At each level check for a langauge specific template.
+    template_sets = [check_for_language_specific_template(template_set) for template_sets in template_sets]
```

Comment:
I think this is worse for number of DB hits.  4 here and then worst case 4 more for the most common case (using the base template)